options/ansi/generic/file-io.cpp

```diff
@@ -584,7 +584,7 @@ int fileno_unlocked(FILE *file_base) {
 
 int fileno(FILE *file_base) {
 	auto file = static_cast<mlibc::fd_file *>(file_base);
-	frg::unique_lock<FutexLock> lock(file->_lock);
+	frg::unique_lock<RecursiveFutexLock> lock(file->_lock);
```

Comment:
It should actually be possible to drop the template args completely due to CTAD.